packages/babel-parser/src/util/scope.js

```diff
@@ -89,55 +93,79 @@ export default class ScopeHandler {
   }
 
   declareName(name: string, bindingType: ?BindingTypes, pos: number) {
-    let redeclared = false;
-    if (bindingType === BIND_LEXICAL) {
-      const scope = this.currentScope();
-      redeclared =
-        scope.lexical.indexOf(name) > -1 ||
-        scope.functions.indexOf(name) > -1 ||
-        scope.var.indexOf(name) > -1;
-      scope.lexical.push(name);
-      if (this.inModule && scope.flags & SCOPE_PROGRAM) {
-        this.undefinedExports.delete(name);
-      }
-    } else if (bindingType === BIND_SIMPLE_CATCH) {
-      const scope = this.currentScope();
-      scope.lexical.push(name);
-    } else if (bindingType === BIND_FUNCTION) {
+    if (
```

Comment:
The changes in this file are mostly refactoring things in different functions, to make it possible to overwrite them.